src/py/flwr/superexec/exec_servicer.py

```diff
@@ -38,13 +41,14 @@ class ExecServicer(exec_pb2_grpc.ExecServicer):
     def __init__(self, executor: Executor) -> None:
         self.executor = executor
         self.runs: Dict[int, RunTracker] = {}
+        self.lock = threading.Lock()
```

Comment:
Can you elaborate on why we need a global lock (global in the context of all runs)?

Reply:
Good question. The idea behind this is so that only the `capture_thread` that acquires `self.lock` can at any one time append to the `.logs` member of the `.runs[run.run_id]` dictionary. The reason was to ensure thread-safe updates to the dictionary (which I wasn't sure about). 

But, I dug deeper and found from [this SO discussion](https://stackoverflow.com/a/54681070) that updating Python dictionaries are thread safe if the keys are builtin hashable types, which in this case it is since we use signed int64 for `run.run_id`. So, I can also remove `self.lock` and the `with self.lock` lines - the log dictionary will be updated by their respective `capture_thread`.